src/Persistent/Pure.hs

```diff
@@ -606,6 +608,31 @@ addDelegation = addDb dbDelegationMap
 deleteDelegation :: AUID Delegation -> AUpdate ()
 deleteDelegation did = dbDelegationMap . at did .= Nothing
 
+delegationScopeTree :: User -> Query (Tree DScopeFull)
+delegationScopeTree user = unfoldTreeM discover DScopeGlobalFull
+  where
+    discover :: DScopeFull -> Query (DScopeFull, [DScopeFull])
+    discover DScopeGlobalFull = do
+        schoolIdeas <- DScopeIdeaFull <$$> findWildIdeasBySpace SchoolSpace
+        let ideaSpaces = DScopeIdeaSpaceFull . ClassSpace <$> (user ^.. userSchoolClasses)
```

Comment:
moderator should be able to see all classes.  we have that functionality somewhere, let me find it...